### DISCUSSION TITLE
how make :set permanent?

### DISCUSSION THREAD

**Discussion by kurokirasama** [qutebrowser/qutebrowser#8286] [Q&A] [answered]
Hey, I just found out and installed the browser. I can change some settings with `:set`, but the changes aren't kept when the browser restart...

**Comment by mschilli87** [comment]
Are you using a `config.py` or the auto config? Could you share the output of `:version`?

**Comment by kurokirasama** [comment]
Hey, the config directory has a `config.py` file but is empty.
The version is qutebrowser v2.5.4. I installed via the package manager in ubuntu 24.04

**Reply by mschilli87** [reply]
The output of `:version` contains much more information that could be useful.
My guess is that your empty `config.py` is picked up and takes precedence over the autoconfig.
So I'd try either removing that empty config or actually loading the autoconfig from within your `config.py`.

**Reply by The-Compiler** [reply]
Yep, if you have a `config.py`, you need to decide if/where to load settings made via the UI. See the [docs](https://github.com/qutebrowser/qutebrowser/blob/main/doc/help/configuring.asciidoc):

> For more advanced configuration, you can write a `config.py` file - see ["Configuring qutebrowser via config.py"](https://github.com/qutebrowser/qutebrowser/blob/main/doc/help/configuring.asciidoc#configpy). When a `config.py` exists, the `autoconfig.yml` file **is not read anymore** by default. You need to [load it from `config.py`](https://github.com/qutebrowser/qutebrowser/blob/main/doc/help/configuring.asciidoc#configpy-autoconfig) if you want settings changed via `:set`/`:bind` to persist between restarts.

What's confusing me is that qutebrowser should display a warning about this on every start, unless you're using `--temp-basedir`:

> Errors occurred while reading config.py: 
>
> autoconfig loading not specified: Your config.py should call either `config.load_autoconfig()` (to load settings configured via the GUI) or `config.load_autoconfig(False)` (to not do so) 

Are you not getting that warning? I just tried with v2.5.4 and an empty `config.py` and it works fine.

**Reply by emanuele6** [reply, marked as answer]
If you have an empty `config.py`, `qutebrowser` will not load `autoconfig.yml` automatically (the file that contains among other things the settings you have set with `:set`).

You need to make sure to call `config.load_autoconfig()` at the top of `config.py`.

**Reply by kurokirasama** [reply]
thanks a lot! Little question, can i export the current config into a config.py file?

**Reply by The-Compiler** [reply]
Yes, see the [`:config-write-py` command](https://qutebrowser.org/doc/help/commands.html#config-write-py)
